src/lib/plug.ts

```diff
@@ -1,29 +1,124 @@
-import { deserializers, serializers } from "./conversions";
+import { Accessory } from "./accessory";
 import { IPSODevice } from "./ipsoDevice";
-import { deserializeWith, ipsoKey, serializeWith } from "./ipsoObject";
+import { doNotSerialize, ipsoKey, IPSOOptions } from "./ipsoObject";
+
+export type PlugOperation = Partial<Pick<Plug,
+	"onOff"
+	>>;
 
 /* istanbul ignore next */
 export class Plug extends IPSODevice {
 
-	@ipsoKey("5805")
-	public cumulativeActivePower: number = 0.0; // <float>
+	constructor(options?: IPSOOptions, accessory?: Accessory) {
+		super(options);
+
+		// In order for the simplified API to work, the
+		// accessory reference must be a proxy
+		if (accessory != null && !accessory.isProxy) {
+			accessory = accessory.createProxy();
+		}
+		this._accessory = accessory;
+
+		// get the model number to detect features
```

Comment:
Not sure if we need this part. It was used in lights to determine if lights support RGB or not.
Maybe if they release dimmable plugs (is that even a thing?) in the future.